pkg/worker/verifier/transactional_producer_worker.go

```diff
@@ -0,0 +1,345 @@
+package verifier
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/rcrowley/go-metrics"
+	"github.com/redpanda-data/kgo-verifier/pkg/util"
+	worker "github.com/redpanda-data/kgo-verifier/pkg/worker"
+	log "github.com/sirupsen/logrus"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"golang.org/x/sync/semaphore"
+)
+
+type TransactionalProducerConfig struct {
+	workerCfg          worker.WorkerConfig
+	name               string
```

Comment:
WorkerConfig already has a name attribute (it's a piece of tech debt that the other producer code  had its own as well)

Reply:
Will switch over to setting the transactional ID to use this.